codegen/src/main/kotlin/org/jetbrains/bsp/generators/ir/IR.kt

```diff
@@ -59,21 +66,31 @@ sealed interface EnumType<A> {
 
 data class EnumValue<A>(val name: String, val value: A, val hints: List<Hint>)
 
-sealed interface Type {
-    data class Set(val member: Type) : Type
-    data class List(val member: Type) : Type
-    data class Map(val key: Type, val value: Type) : Type
-    data class Ref(val shapeId: ShapeId) : Type
+sealed interface InnerType {
+    data class Set(val member: Type) : InnerType
+    data class List(val member: Type) : InnerType
+    data class Map(val key: Type, val value: Type) : InnerType
+    object Ref : InnerType
 
     // Should be data objects
-    object Unit : Type
-    object Bool : Type
-    object String : Type
-    object Int : Type
-    object Long : Type
-    object Json : Type
+    object Unit : InnerType
+    object Bool : InnerType
+    object String : InnerType
+    object Int : InnerType
+    object Long : InnerType
+    object Json : InnerType
 }
 
+data class Type(val shapeId: ShapeId, val type: InnerType) {
```

Comment:
Type zmieniłbym na IrShape, a InnerType na Type, chyba tak miałoby więcej sensu. Co sądzisz?

Reply:
też mi to aktualne nazewnictwo nie do końca odpowiada więc mogę tak zmienić

Reply:
done